public/apps/account/password-reset-panel.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiIcon,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiOverlayMask,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import { FormRow } from '../configuration/utils/form-row';
+import { AppDependencies } from '../types';
+import { getUserDetail, updateUser } from '../configuration/utils/internal-user-detail-utils';
+import { InternalUserUpdate } from '../configuration/types';
+import { buildUrl } from '../configuration/utils/url-builder';
+
+interface PasswordResetPanelProps extends AppDependencies {
+  username: string;
+  handleClose: () => void;
+}
+
+export function PasswordResetPanel(props: PasswordResetPanelProps) {
+  const [currentPassword, setCurrentPassword] = useState<string>('');
+  // reply on backend response of login call to verify
+  const [isCurrentPasswordInvalid, setIsCurrentPasswordInvalid] = useState<boolean>(false);
+
+  const [newPassword, setNewPassword] = useState<string>('');
+  // reply on backend response of user update call to verify
+  const [isNewPasswordInvalid, setIsNewPasswordInvalid] = useState<boolean>(false);
+
+  const [repeatNewPassword, setRepeatNewPassword] = useState<string>('');
+  const [isRepeatNewPasswordInvalid, setIsRepeatNewPasswordInvalid] = useState<boolean>(false);
+
+  const handleReset = async () => {
+    // validate the current password
+    try {
+      await props.coreStart.http.post('/auth/login', {
+        body: JSON.stringify({
+          username: props.username,
+          password: currentPassword,
+        }),
+      });
+    } catch (e) {
+      setIsCurrentPasswordInvalid(true);
+      return;
+    }
+
+    // update new password
+    try {
+      const user = await getUserDetail(props.coreStart.http, props.username);
+
+      const updateObject: InternalUserUpdate = {
+        password: newPassword,
+        backend_roles: user.backend_roles,
+        attributes: user.attributes,
+      };
```

Comment:
I think here need to use `POST api/v1/configuration/account` to update password. The InternalUserUpdate only works for security manager.

Reply:
Sure. Will switch.